src/features/feedback/service.ts

```diff
@@ -0,0 +1,58 @@
+//import { ServiceResult } from "@/services";
+
+import type { Feedback, ServiceResult } from "./types";
+import { ref } from "vue";
+
+export function useFeedbackService() {
+  if (!window.gatewayBaseUri) {
+    console.error("contactmomentenBaseUri missing");
+  }
+
+  const feedbackUrl = window.gatewayBaseUri + "/reviews";
+
+  const postFeedback = (data: Feedback) => {
+    const state = ref(<ServiceResult>{ loading: true, error: false });
+
+    const promise = fetch(feedbackUrl, {
+      method: "POST",
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(MapModel(data)),
+    })
+      .then((r) => {
+        if (!r.ok) {
+          state.value.error = true;
+          throw new Error();
+        }
+        state.value.success = true;
+        return r.json();
+      })
+      .catch(() => {
+        state.value.error = true;
+      })
+      .finally(() => {
+        state.value.loading = false;
+      });
+
+    const result = { state: state, promise: promise };
```

Comment:
Dit is een beetje krom. Als het ServiceResult teveel in de weg zit, dan maar niet gebruiken, maar een nettere oplossing met met alleen een ServiceResult lijkt me ook haalbaar

Reply:
aangepast in overleg